integrationTests/activities-parsing.integration.spec.ts

```diff
@@ -165,7 +192,13 @@ describe('Integration tests with native & custom dom parser', () => {
             expect(event.getFirstActivity().getSquashedStreamData(DataSpeed.type).length).toEqual(expectedSamplesLength);
             expect(event.getFirstActivity().getSquashedStreamData(DataCadence.type).length).toEqual(expectedSamplesLength);
             expect(event.getFirstActivity().getSquashedStreamData(DataHeartRate.type).length).toEqual(expectedSamplesLength);
-            expect(event.getFirstActivity().getSquashedStreamData(DataPace.type).length).toEqual(expectedSamplesLength);
+            expect(event.getFirstActivity().getDuration().getValue()).toEqual(expectedElapsedTime);
+            const movingTime = (<DataMovingTime>event.getFirstActivity().getStat(DataMovingTime.type)).getValue()
+            expectTolerance(movingTime, expectedMovingTime, expectedMovingTime * MOVING_TIME_TOLERANCE);
+
+            // Remove because the number you have looks its from stravas data
+            // expectTolerance(<number>event.getFirstActivity().getStats().get(DataAscent.type)?.getValue(), expectedElevationGain, ASCENT_ELEVATION_TOLERANCE);
```

Comment:
Indeed but both strava & garmin are reporting **310m elevation gain**:
- https://connect.garmin.com/modern/activity/4108490848
- https://www.strava.com/activities/2749884297

Sports-lib reports **411m** :(

Note: In elevate i applied this smoothing to altitude streams before my calculations (elevation gain included):

```typescript
protected smoothAltitudeStream(altitudeStream: number[]): number[] {
	const kalman = new KalmanFilter();
	altitudeStream = altitudeStream.map(elevation => kalman.filter(elevation));
	const lowPassFilter = new LowPassFilter(0.2);
	return lowPassFilter.smoothArray(altitudeStream);
}
```
After analysis it gave me elevation gain results very close to strava and garmin values.

I attempted to modify the `getEventDataTypeGainOrLoss` static method to get this behavior inside, it gives better results on the elevation gain assertions:

```typescript
  private static getEventDataTypeGainOrLoss(
    activity: ActivityInterface,
    streamType: string,
    gain: boolean,
    startDate?: Date,
    endDate?: Date,
    minDiff?: number): number {
    let streamData = activity.getSquashedStreamData(streamType, startDate, endDate);
    const KalmanFilter = require('kalmanjs');
    const kalman = new KalmanFilter();
    streamData = streamData.map(elevation => kalman.filter(elevation));
    streamData = <number[]> new LowPassFilter(0.2).smoothArray(streamData);
    return this.getGainOrLoss(streamData, gain, minDiff);
  }
```
But this could provide impacts & regressions on others sports-lib calculations? So no commits. Just experimentation :)

Reply:
Yes that is correct. That is not how the ascent etc should be solved